lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -7,96 +7,77 @@
 #include <stdio.h>
 #include <string.h>
 #include <zephyr.h>
-#include <modem/at_cmd.h>
+#include <nrf_modem_at.h>
 #include <modem/at_cmd_parser.h>
 #include <modem/at_params.h>
 #include <nrf_modem_limits.h>
 #include <modem/modem_key_mgmt.h>
 #include <logging/log.h>
 
-#define MODEM_KEY_MGMT_OP_LS "AT%CMNG=1"
-#define MODEM_KEY_MGMT_OP_RD "AT%CMNG=2"
-#define MODEM_KEY_MGMT_OP_WR "AT%CMNG=0"
-#define MODEM_KEY_MGMT_OP_RM "AT%CMNG=3"
-
 #define AT_CMNG_PARAMS_COUNT 5
 #define AT_CMNG_CONTENT_INDEX 4
 
+#define ENABLE 1
+#define DISABLE 0
+
 LOG_MODULE_REGISTER(modem_key_mgmt, CONFIG_MODEM_KEY_MGMT_LOG_LEVEL);
 
 static char scratch_buf[4096];
 
-static int cmee_is_active(void)
+static bool cmee_is_active(void)
 {
 	int err;
-	char response[sizeof("+CMEE: X\r\n")];
+	int active;
 
-	err = at_cmd_write("AT+CMEE?", response, sizeof(response), NULL);
-	if (err) {
+	err = nrf_modem_at_scanf("AT+CMEE?", "+CMEE: %d", &active);
+	if (err < 0) {
 		return err;
```

Comment:
The function was changed to return a boolean, but tries to return an error here.

Reply:
Fixed